### DISCUSSION TITLE
observed PK profile as inputs to drive PD model

### DISCUSSION THREAD

**Discussion by mark0935git** [metrumresearchgroup/mrgsolve#815] [Q&A] [answered]
Hi Mrgsolve developer,

I wish to know whether mrgsolve allows to take observed or pre-simulated PK profiles as inputs and use them to drive the PD model or not?
So the mrgsolve model will only parameters and ODEs of PD model that is drived by input observed PK profiles.

If this is possible, how should we code such model? Do we need to input the observed PK profiles as covariates that vary over time?

**Comment by kylebaron** [comment]
Hi @mark0935git - 

I moved this to our "Discussion" area which works just like issues but is better suited for Q&A type stuff. 

Yes, that's right - you can bring the exposures into the problem like a time-varying covariate and use them to drive the PD effect.  Remember that mrgsolve uses the next observation carry backward approach .... so it would be the exposure on the record that the system is advancing **to** that will drive the advance.  Also, assuming that the PD is modeled by some ODE, keep in mind that exposure would nevertheless be "constant" over the time that the PD model is getting integrated.  So if you want PK to evolve as the PD evolves, that would have to be simulated in the model too.

Let me know if I can help with anything else!

Thanks,
Kyle

**Comment by riggsmm** [comment, marked as answer]
Hi @mark0935git , 
To add to @kylebaron's suggestion, you could also consider a semi-parametric approach to interpolate the PK values between observation records (as an alternate to the step-function default).  The attached control file (developed for NONMEM) should transfer pretty easily into mrgsolve code if you're interested in trying that approach.  This will then provide a continuous prediction of the PK using the solver time (noting that "T" in NONMEM is "SOLVERTIME" in the .cpp (mrgsolve) file).

The interpolation works best when observations are closer in time though, in general, it's usually an improvement over next observation carry backward.  

[nonpar_pkpd.ctl.txt](https://github.com/metrumresearchgroup/mrgsolve/files/6353752/nonpar_pkpd.ctl.txt)

For further reading on this approach: Kowalski, K. G., & Karim, A. (1995). A semicompartmental modeling approach for pharmacodynamic data assessment. Journal of Pharmacokinetics and Biopharmaceutics, 23(3), 307–322.

Hope this helps! 

Kindly,
Matt

**Reply by mark0935git** [reply]
Thanks all your answers! I have resolved the issue following your instructions!
